filters/auth/authclient.go

```diff
@@ -42,9 +43,9 @@ func newAuthClient(baseURL string, timeout time.Duration) (*authClient, error) {
 	return &authClient{url: u, client: client, quit: quit}, nil
 }
 
-func (ac *authClient) getTokenintrospect(token string, ctx filters.FilterContext) (tokenIntrospectionInfo, error) {
+func (ac *authClient) getTokenintrospect(token string, ctx filters.FilterContext, clientId string, clientSecret string) (tokenIntrospectionInfo, error) {
 	info := make(tokenIntrospectionInfo)
-	err := jsonPost(ac.url, token, &info, ac.client, ctx.Tracer(), ctx.ParentSpan(), tokenIntrospectionSpan)
+	err := jsonPost(ac.url, token, clientId, clientSecret, &info, ac.client, ctx.Tracer(), ctx.ParentSpan(), tokenIntrospectionSpan)
```

Comment:
please use `ac.url.User` to set the UserInfo, see https://golang.org/pkg/net/url/#URL and https://golang.org/pkg/net/url/#UserPassword, like this you do not need to pass 2 additional arguments around